app/domain/authentication/authn_oidc/v2/data_objects/authenticator.rb

```diff
@@ -47,6 +69,26 @@ def name
           def resource_id
             "#{account}:webservice:conjur/authn-oidc/#{service_id}"
           end
+
+          # Returns the validity duration, in seconds, of an instance's access tokens.
+          def token_ttl
+            return TOKEN_TTL_DEFAULT unless @token_ttl
+
+            begin
+              duration = ActiveSupport::Duration.parse(@token_ttl) if @token_ttl
+            rescue ActiveSupport::Duration::ISO8601Parser::ParsingError
```

Comment:
I'm in the process of adding a "Status" check for the new OIDC. I think we should move the ISO8601 check into that verification.

Reply:
Good idea, the Status check should definitely be able to verify the custom TTL config. Given that the `Status` and `TokenFactory` classes both need to decode ISO 8601 strings, does it make sense to keep that conversion in a method on the `Authenticator` class?

Related comment from above:
> Instead of forcing the Annotation class to have an awareness of a Conjur token (and that token's TTL), let's push all this into the TokenFactory class.

The `Authenticator` class needs some concept of a custom TTL, even if only in its ISO 8601 form. Maybe it's best to move the max/min range into the `TokenFactory`, but keep ISO 8601 parsing as part of the `Authenticator` class.

Reply:
Given the edge cases we were talking about, @john-odonnell, with DST changes and leap seconds and all that, if we do have to add code to handle those, having one place do the processing seems like a really good idea too.